internal/host/logscmd.go

```diff
@@ -69,17 +72,66 @@ func (i *logsCmd) getNonBootstrapMastersIPsInHostCluster(ctx context.Context, ho
 		return nil, err
 	}
 
+	if swag.BoolValue(cluster.UserManagedNetworking) {
+		return i.getHostsIps(cluster)
+	}
+	return i.getHostsIpsfromMachineCIDR(cluster)
+}
+
+func (i *logsCmd) getHostsIpsfromMachineCIDR(cluster common.Cluster) ([]string, error) {
 	var ips []string
 	for _, h := range cluster.Hosts {
 		if h.Bootstrap || h.Role == models.HostRoleWorker {
 			continue
 		}
 		ip, err := network.GetMachineCIDRIP(h, &cluster)
 		if err != nil {
-			i.log.WithError(err).Errorf("failed to get machine cidr IP for host %s", host.ID)
+			i.log.WithError(err).Errorf("failed to get machine cidr IP for host %s", h.ID)
 			return nil, err
 		}
 		ips = append(ips, ip)
 	}
 	return ips, nil
 }
+
+func (i *logsCmd) getHostsIps(cluster common.Cluster) ([]string, error) {
+	var ips []string
+	for _, h := range cluster.Hosts {
+		if h.Bootstrap || h.Role == models.HostRoleWorker {
+			continue
+		}
+		var inventory models.Inventory
+		err := json.Unmarshal([]byte(h.Inventory), &inventory)
+		if err != nil {
+			i.log.WithError(err).Warn("Fail to getHostsIps:  Inventory parse")
+			return nil, err
+		}
+		for _, intf := range inventory.Interfaces {
+			ips, err = i.parseIps(intf.IPV4Addresses, ips)
```

Comment:
NIT: since we send ips slice in order to append and return it, maybe its better to use in-place append
```
func dolist(ips *[]string) error {
	*l = append(*l, newip)
```

Reply:
Fixed, thanks